mega-linter-runner/generators/mega-linter/templates/mega-linter.yml

```
@@ -1,36 +1,116 @@
-# MegaLinter GitHub Action configuration file
-# More info at https://megalinter.io
+##
+# MegaLinter GitHub Action configuration file.
+#
+# MegaLinter is an Open-Source tool for CI/CD workflows that analyzes the
+# consistency of your code, IAC, configuration, and scripts in your repository
+# sources, to ensure all your projects sources are clean and formatted whatever
+# IDE/toolbox is used by their developers, powered by OX Security.
+#
+# More info at:
+# https://megalinter.io
+#
 ---
 name: MegaLinter
 
-# Trigger mega-linter at every push. Action will also be visible from
-# Pull Requests to main
+##
+# By default, Megalinter runs whenever a Pull Request is opened with the default
+# branch, or on any push.
+#
+# Later logic enforces a full code-wide test on only the `production` and
+# `staging` Branches. The default Branch only has changed files linted for
+# efficiency.
+#
 on:
-  # Comment this line to trigger action only on pull-requests
-  # (not recommended if you don't pay for GH Actions)
+
+  ##
+  # To only have MegaLinter run on Pull Requests, comment out the following line.
+  # It is not recommended to use `push` linting on all pushes (the default,
+  # here)if you do not pay for GitHub Actions.
+  #
+  # Some projects also prefer to limit push linting to specific branches, such
+  # as when a `production` branch is updated. This can be done by adding a
+  # `branches` filter, here, such as:
+  #
+  # push:
+  #   branches:
+  #     - production
+  #
   push:
 
+  ##
+  # Run whenever a Pull Request occurs on the configured, default branch.
```

```suggestion
  # Run whenever a Pull Request occurs on the configured default branch.
```
Why was there a comma?